tools/github_helper/main.go

```diff
@@ -309,21 +310,30 @@ func (h helper) updatePullRequest(pr *github.PullRequest, s *github.CombinedStat
 		return false
 	}
 
-	if (state == gh.failure || state == gh.error) && (len(h.CheckToSkip) > 0) {
-		isSuccess := true
-		for _, status := range s.Statuses {
-			if *status.State == gh.error || *status.State == gh.failure {
-				if skipContext(*status.Context) {
-					continue
-				} else {
-					isSuccess = false
-					break
-				}
+	// Not trusting Combined output.
+	// Not counting successes as there could be no required checks.
+	failures := 0
+	pending := 0
+	for _, status := range s.Statuses {
+		if *status.State == gh.error || *status.State == gh.failure {
+			if skipContext(*status.Context) {
+				continue
 			}
+			failures++
```

Comment:
I think it is better if we have a full view of it. I ll had a logging statement to help us debug this.